pkg/vm/manager.go

```diff
@@ -0,0 +1,404 @@
+package vm
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"strconv"
+	"syscall"
+	"time"
+
+	firecracker "github.com/firecracker-microvm/firecracker-go-sdk"
+	"github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg"
+)
+
+const (
+	// FCBin  path for firecracker
+	FCBin = "/bin/firecracker"
+	// JailerBin path for fc jailer
+	JailerBin = "/bin/jailer"
+	// FCSockDir where vm firecracker sockets are kept
+	FCSockDir = "/var/run/firecracker"
+
+	defaultKernelArgs = "ro console=ttyS0 noapic reboot=k panic=1 pci=off nomodules"
+)
+
+// vmModuleImpl implements the VMModule interface
+type vmModuleImpl struct {
+	root string
+}
+
+var (
+	_ pkg.VMModule = (*vmModuleImpl)(nil)
+
+	errScanFound = fmt.Errorf("found")
+)
+
+// NewVMModule creates a new instance of vm manager
+func NewVMModule(root string) (pkg.VMModule, error) {
```

Comment:
It is better to return a actual type instead of an interface.
```suggestion
func NewVMModule(root string) (*vmModuleImpl, error) {
```

Reply:
I understand, but this is a private type. To make this possible I will have to make it a public type . like `VMModule` but then, it will be possible to developers to create an instance of this type directly without calling the `NewVMModule` method!

I am not sure which is worse actually.

Reply:
Indeed you're right, being a private type if is better to just return the interface then. My bad.